tests/test_operation2.py

```diff
@@ -362,75 +360,32 @@ def test_operation_composed_definition(api):
     assert operation.method == 'GET'
     assert operation.produces == ['application/json']
     assert operation.consumes == ['application/json']
-    assert operation.security == [{'oauth': ['uid']}]
+
     expected_body_schema = op_spec["parameters"][0]["schema"]
     expected_body_schema.update({'definitions': DEFINITIONS})
     assert operation.body_schema == expected_body_schema
 
 
-def test_operation_local_security_oauth2(api):
+def test_operation_local_security_oauth2(security_handler_factory):
     verify_oauth = mock.MagicMock(return_value='verify_oauth_result')
-    api.security_handler_factory.verify_oauth = verify_oauth
+    security_handler_factory.verify_oauth = verify_oauth
 
-    op_spec = make_operation(OPERATION8)
-    operation = Swagger2Operation(api=api,
-                                  method='GET',
-                                  path='endpoint',
-                                  path_parameters=[],
-                                  operation=op_spec,
-                                  app_produces=['application/json'],
-                                  app_consumes=['application/json'],
-                                  app_security=[],
-                                  security_definitions=SECURITY_DEFINITIONS_LOCAL,
-                                  definitions=DEFINITIONS,
-                                  parameter_definitions=PARAMETER_DEFINITIONS,
-                                  resolver=Resolver())
-    assert isinstance(operation.function, types.FunctionType)
-    security_decorator = operation.security_decorator
-    assert len(security_decorator.args[0]) == 1
-    assert security_decorator.args[0][0] == 'verify_oauth_result'
-    verify_oauth.assert_called_with(math.ceil, api.security_handler_factory.validate_scope, ['uid'])
+    SecurityOperation(security_handler_factory=security_handler_factory,
+                      security=[{'oauth': ['uid']}],
+                      security_schemes=SECURITY_DEFINITIONS_LOCAL)
 
-    assert operation.method == 'GET'
```

Comment:
Use `SecurityOperation` and remove intermediate checks not related to security for tests related to security.